### ISSUE TITLE
Enable easy data joins for polygon mapping by indicating official concordance value (with backfill)

### ISSUE THREAD

**Issue by nvkelso** [whosonfirst-data/whosonfirst-data#2164]
![image](https://github.com/whosonfirst-data/whosonfirst-data/assets/853051/bf2da01c-4641-488a-a9a6-0776440cf5bb)

To facilitate thematic [choropleth maps](https://en.wikipedia.org/wiki/Choropleth_map) for data visualization (dataViz), polygon features need to indicate which of our many concordance values is the "official" concordance for that country's national statistical (census) agency. There should be a single value for this purpose (and an additional new property to list other keys that are "official" in other ways).

To backfill these new properties, the original Quattroshapes import includes properties to mine for these values for around 40 countries, and more recent localadmin imports similarly have properties to mine (so too does Mesoshapes). Ideally we add new WOF sources for each of these countries, and default to `qs:local_id` or `meso:local_id` when that's tricky. For a limited set of countries, we also have more recent official data we've imported that sometimes has the code, or can be used to verify the code. In some cases like Finland we imported the geometry but it didn't include a concordance. In the USA, the existing `uscensus:geoid` concordances already exist, and the new "official concordance key" property is the only work needed to specific the mapping (while many of the other countries need concordance values also set).

Generally, the "census" / "statistical" IDs should be given priority for the default official value – less so the GIS unique feature IDs from the national mapping agency (which are more like FIDs either numerical or GUIDs. Often the national mapping agency includes these unique and stable concordance IDs on their files (in addition to their own GUID), and sometimes not. 

In rare cases, there may be more than one official code. For example, in France they have a `fra-insee:code` (default)
and `frgov:epci_code` (alt) for communes, because they sometimes group together into a type of regional federation and data is distributed in both ways.

Ideally the official concordance value is already a "qualified" code – which is mostly true. But there are exceptions like in Spain: `esp-ine:code` (default) and `esp-ine:code_qualified` where it includes the parent's national ID code as a prefix, which is more how we'd expect things in the US but not always how data is published elsewhere. Poland is another an exception.

There are other example where a regional body like EuroGeographics will set a multicountry ID (mostly by prefixing the country code and a `_` separator) onto the national ID. So those _could_ also go into an "alt" listing.

So far I've seen 1 example where there are 2 different official IDs in a country. France has fra-insee:code (default)
and frgov:epci_code (alt) for communes, because they sometimes group together into a type of regional federation.

Generally I think the "census" / "statistical" IDs should be given priority for the default official value – less so the GIS unique feature IDs from the national mapping agency (which are more like FIDs either numerical or GUIDs and I've generally avoided adding as they are rare and not useful for demographic data joins, you'll see the GUID type IDs in the United Kingdom).

Sometimes I've also researched a "qualified" code in addition to the basic code. In Spain: esp-ine:code (default) and esp-ine:code_qualified where it includes the parent's national ID code as a prefix (well, it's there but just at province level not also the parent autonomous community), which is more how we'd expect things in the US but not always how data is published elsewhere. NOTE: Poland is an exception in that I couldn't figure out what the basic could would be:pol-gus:terc_code_qualified.

There are other example where a regional body like EuroGeographics will set a multicountry ID (mostly by prefixing the country code and a _ separator) onto the national ID. So those could also go into an "alt" listing.

A note about ISO: Yes that's an "official" _international_ standard for `country`, `dependency`, and `region` level IDs (sometimes `macroregion`, too). So we should endeavor to backfill those `iso:code` concordances as much as possible – but we should prefer the national code over the international code as data for choropleth mapping comes from national statistical agencies not ISO, but it's OK to set an `iso:code` in an alt list (or even as primary if there is no national concordance ID to set). The Statoids HASC IDs are "semi-official" in that they also cover 50% of the `localadmin` level we have, while ISO doesn't cover that level, so same thing goes for HASC as ISO vis-a-vis default versus variant / alt.

There's a lot of good discussion in the https://github.com/whosonfirst/whosonfirst-properties/pull/113 with @thisisaaronland and @stepps00 about what to call the new properties.

**Comment by nvkelso** [comment]
![image](https://github.com/whosonfirst-data/whosonfirst-data/assets/853051/d5140251-c60d-4341-9c13-3d56bcdb5f03)

**At all extant "coarse" placetypes, in the following countries:**

_New official concordance key indicators and value backfills, and also ISO for regions_

- AT
- AU
- BE
- BG
- BR
- CH
- CY
- CZ
- DE
- DK
- EE
- ES
- FI
- FR
- GB
- GL
- GR
- HR
- HU
- IE
- LT
- LV
- MD
- MT
- MX
- NL
- NO
- NZ
- PL
- PT
- RO
- RS
- SE
- SK
- UA
- US

Limited to `county` and ISO for `region` in handful of countries:
- BY
- CN
- JP
- KR
- RU

All other countries just have ISO for `region` (99% coverage) and and HASC for `county` (around 50% coverage).

![image](https://github.com/whosonfirst-data/whosonfirst-data/assets/853051/94f8de3a-99ce-476e-b43d-dc6a03ee7072)

For Localadmin:
- AT
- AU
- BE
- BR
- CH
- DE
- DK
- EE
- ES
- FI
- FR
- GB – missing coverage of Scotland and Northern Ireland
- IS – not available
- IT
- NL
- NO
- NZ – not available
- PL
- SE
- SI
- US

**Further work could be done:**

_Doesn't list countries with just handful of missing concordances._

- AZ - set the macroregion (economic region) official codes
- BR - set the macroregion official codes (these are statistical regions)
- HU - set the macroregion official codes (these are statistical regions)
- RS - set the macroregion official codes for Central Serbia
- TW - set the macroregion official codes (gets a bit house of mirrors here between the WOF placetypes)
- GB - region for ceremonial counties don't have official codes?
- `NULL` - set the region official codes for handful of "countries" that lack the `wof:country` being set because they are points and they fall outside a parent polygon
- AU - set the macrocounty official codes
- GF - set the macrocounty official codes
- `NULL` - set the county official codes for handful of "countries" that lack the `wof:country` being set because they are points and they fall outside a parent polygon
- AE - set the county official codes
- AM - set the county official codes
- AO - set the county official codes
- AT - set the county official codes
- BA - set the county official codes
- CD - set the county official codes
- CA - set the county official codes for handful of major cities
- CH - set the county official codes for couple dozen
- ET - set the county official codes for couple dozen
- FI - set the county official codes for couple dozen
- IE - set the county official codes for Tipperary
- IN - set the county official codes for couple dozen
- IQ - set the county official codes for couple dozen in entire Kurdistan
- KP - set the county official codes
- KZ - set the county official codes for handful
- MG - set the county official codes
- TW - set the county official codes
- TZ - set the county official codes
- UY - set the county official codes

**Comment by nvkelso** [comment]
PRs:

- Sources: https://github.com/whosonfirst/whosonfirst-sources/pull/208
- Properties: https://github.com/whosonfirst/whosonfirst-properties/pull/113
- Example / test case record: https://github.com/whosonfirst-data/whosonfirst-data-admin-us/pull/155
- A few hundred other PRs!

**Comment by nvkelso** [comment]
**My general algo for _within a single country_ placetypes:**

- Prefer national statistical agency IDs
- Fallback to national mapping agency IDs (when they aren't GUIDs)
- Fallback to ISO codes for `country` and `region` (and `macroregion`) – and otherwise set these as `wof:concordances_official_alt`
- Fallback to HASC codes for `region` (and `macroregion`) and `county` (and `macrocounty`) and `localadmin` – and otherwise set these as `wof:concordances_official_alt`

**My country/dependency algo is:**

- Prefer the ISO codes for `country` and `dependency` as the comparable peer set is extraterritorial and ISO manages that instead.

While Statoids is not "official", it is the only other widely available international convention, covering around 50% of the county features in WOF, and since WOF has legal rights to use the HASC codes, it's a preferential WOF source. 

Looking at the case of India, only the HASC codes are currently known to WOF and it'd be a medium sized research project to get the official national codes (assuming they exist).

For a given country I've tried to always prefer the national statistical agency's IDs... but in some odd cases individual records won't have an ID. 

For some countries I was easily able to research what the ID should be (like new US `county` features), but in other cases that was prohibitive, so I backfilled the official with another known ID in our set of 4 preferred and variant sources so it would at least have an ID, and the key of that ID could still be noted as being different than it's peers. 

**More on the country / dependency edge case:**

The US is the only country with data in WOF where the NMA/NSA/Census describes the whole country as having a code. Probably other countries do this too IRL, but I didn't see it in our data. 

For other placetypes the peers are all _in-country_ so using the NMA/NSA/Census makes logical sense.

But for country/dependency placetypes all the peers are _out-of-country_, so it made more sense to me to always use `ISO:code` for that (and in some cases I set a `ISO:code_pseudo` like for Kosovo).

NOTE: Since the _value_ of the United States official geoid and the ISO code are the same (`US`), this amounts to a no-op to end users.
